### PR TITLE
alternative way of deleting files in cubelets directory

### PR DESCRIPTION
**Pull request by vdhulst**
This code reads all files in the cubelets directory and deletes them if the file name matches the base file name. All other files and directories are left untouched. Avoids leaving files in the directory if the numbering for some reason is not consecutive. No assumptions need to be made about the completeness of the files in the directory, it just deletes all files with the same base file name as the ones that need to be (over)written.

Code can be simplified a bit more. Will be the next step.

### CODE REVIEW

File: sofia.c [hunk 1]
```diff
@@ -542,7 +547,35 @@ int main(int argc, char **argv)
...
+            struct dirent *dir; // structure containing the directory entries
+            DIR *d = opendir(Path_get_dir(path_cubelets)); // open the directory
+	    printf("\nopened directory: " "%s\n\n",Path_get_dir(path_cubelets));
+              if(d == NULL)
+                  printf("\nDirectory" "%s\" not found\n\n", Path_get_dir(path_cubelets)); // invalid path
+                else
+                  {
+                    while ((dir = readdir (d)) != NULL ) // was able to read the directory
+                     {
+                       if(dir-> d_type != DT_DIR && strncmp(dir->d_name, base_name, strlen(base_name)) == 0)
```
Comment (SoFiA-Admin) [inline sofia.c:566]: `dir->d_type` and `DT_DIR` appear to be custom extensions to the POSIX standard and are not POSIX-compliant (see https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/dirent.h.html which only defines the `d_ino` and `d_name` members of `DIR`). This line does not currently compile on my computer (using gcc on Debian), as `DT_DIR` is undefined.

The other problem with this line is that it only checks for files starting with `base_name`, so user-generated content such as `<base_name>_overview_plot.eps` would also get deleted.

**Review by SoFiA-Admin — COMMENTED** [review]
I finally found some time to check and test the new implementation of cubelet deletion, but unfortunately I couldn’t get it to work, as it appears to be using features that are not POSIX-compliant (please find more detailed comments in the source code).

**Comment by RussellJurek** [comment]
Hi Thijs,
         another way to do this is to use C pipes. You can create a pipe
and use it to collect the output of the 'ls -d
base_name<cubelet_name_template>' command. Then you can loop through the
entries and delete them.

Here's a reference for C pipes.

https://www.gnu.org/software/libc/manual/html_node/Creating-a-Pipe.html

Sorry I missed this week's meeting. My travel schedule is as chaotic as I
expected. Boston -> Palmer -> Syracuse and now Toronto. Then on to San
Francisco and San Jose before returning home. I should be on next Monday's
call.

Cheers,
Russell

**Comment by SoFiA-Admin** [comment]
Thanks, @RussellJurek, for that suggestion. I would generally advise against this approach, though, as there is no guarantee that the `ls` command does what you expect it to do, so you might end up wreaking havoc.

**Comment by vdhulst** [comment]
I was not aware that I used POSIX extensions that not all systems support. I resolved this by replacing this with basic file information that stat.h provides. The main goal was to avoid deleting directories as that is potentially dangerous. 

@RussellJurek, I agree with @SoFiA-Admin that using system commands is potentially dangerous and my goal was to work around that as much as possible. I think I managed.

How to avoid removing files that the user may have generated using the same base-name can be avoided by explicitly only removing the files which SoFiA generates. I will look into that.

**Comment by vdhulst** [comment]
this problem has been resolved in a slightly different way by Tobias Westmeier and this version is not needed any more.

**Pull request closed without merging**